lib/stsci/tools/bitmask.py

```diff
@@ -8,18 +9,399 @@
+        or a Python list of bit flags, bit-flipping can be controlled through
+        `flip_bits` parameter.
+
+    flip_bits : bool, None
+        Indicates whether or not to flip the bits of the returned bitmask
+        obtained from input bit flags. This parameter must be set to `None`
+        when input `bit_flags` is either `None` or a Python list of flags.
+
+    Returns
+    -------
+    bitmask : int or None
+        Returns and integer bit mask formed from the input bit value
+        or `None` if input `bit_flags` parameter is `None` or an empty string.
+        If input string value was prepended with '~' (or `flip_bits` was
+        set to `True`), then returned value will have its bits flipped
+        (inverse mask).
+
+    Examples
+    --------
+        >>> from stsci.tools.bitmask import interpret_bit_flags
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags(28))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('4,8,16'))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('~4,8,16'))
+        '1111111111100011'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags('~(4+8+16)'))
+        '1111111111100011'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags([4, 8, 16]))
+        '0000000000011100'
+        >>> "{0:016b}".format(0xFFFF & interpret_bit_flags([4, 8, 16], flip_bits=True))
+        '1111111111100011'
+
+    """
+    has_flip_bits = flip_bits is not None
+    flip_bits = bool(flip_bits)
+    allow_non_flags = False
+
+    if isinstance(bit_flags, int):
+        return (~bit_flags if flip_bits else bit_flags)
+
+    elif bit_flags is None:
+        if has_flip_bits:
+            raise TypeError(
+                "Keyword argument 'flip_bits' must be set to 'None' when "
+                "input 'bit_flags' is None."
+            )
+        return None
+
+    elif hasattr(bit_flags, '__iter__'):
+        pass
+
+    elif isinstance(bit_flags, six.string_types):
+        if has_flip_bits:
+            raise TypeError(
+                "Keyword argument 'flip_bits' is not permitted for "
+                "comma-separated string lists of bit flags. Prepend '~' to "
+                "the string to indicate bit-flipping."
+            )
+
+        bit_flags = str(bit_flags).strip()
+
+        if bit_flags.startswith('~'):
+            flip_bits = True
+            bit_flags = bit_flags[1:].lstrip()
+        else:
+            flip_bits = False
+
+        if bit_flags.startswith('('):
+            if bit_flags.endswith(')'):
+                bit_flags = bit_flags[1:-1].strip()
+            else:
+                raise ValueError('Unbalanced parantheses or incorrect syntax.')
+
+        if ',' in bit_flags:
+            bit_flags = bit_flags.split(',')
+
+        elif '+' in bit_flags:
+            bit_flags = bit_flags.split('+')
+
+        elif bit_flags.upper() in ['', 'NONE', 'INDEF']:
```

Comment:
Shouldn't this be checked in the beginning? Otherwise something like `'~'` or `'~None'` wouldn't raise the exception. Shouldn't be an expected input but it would be surprising if it passed.